gpii/node_modules/contextManager/src/ContextManager.js

```diff
@@ -177,6 +177,7 @@
                 delete finalPayload.lifecycleInstructions;
                 $.extend(true, session, finalPayload);
             }
+            gpii.pcpInterface.sendUserMessage("New configuration driven by context changes");
```

Comment:
When doing a trace with node-inspector I saw that gpii was available at that moment in the closure. Sorry if it's not the way to do it :S 
I will apply your last changes and part from there